src/main/java/com/blackduck/integration/detect/configuration/DetectProperties.java

```diff
@@ -1855,6 +1847,18 @@ private DetectProperties() {
 
     //#region Deprecated Properties
 
+    public static final BooleanProperty DETECT_PROJECT_CODELOCATION_UNMAP =
+        BooleanProperty.newBuilder("detect.project.codelocation.unmap", false)
+            .setInfo("Unmap All Other Scans for Project", DetectPropertyFromVersion.VERSION_4_0_0)
+            .setHelp("If set to true, unmaps all other scans mapped to the project version produced by the current run of Detect.")
+            .setGroups(DetectGroup.PROJECT, DetectGroup.PROJECT_SETTING)
+            .setCategory(DetectCategory.Advanced)
+            .setDeprecated(
+                "This property has been deprecated.",
+                DetectMajorVersion.ELEVEN
```

Comment:
Assuming this is the pattern we follow for other deprecations (haven't personally done or reviewed an MR for a previous deprecation), but should this version be the version we deprecate or the version we will officially remove?

Reply:
Do you know if .setDeprecated() is used to populate https://documentation.blackduck.com/bundle/detect/page/properties/deprecated-properties.html?

Reply:
> Assuming this is the pattern we follow for other deprecations (haven't personally done or reviewed an MR for a previous deprecation), but should this version be the version we deprecate or the version at which we will officially remove?

This refers to the version in which we will remove the property.
With this change, the following is part of Detect log output when a user attempts to use this property: `This property has been deprecated. This property will be removed in 11.0.0.`.

> Do you know if .setDeprecated() is used to populate https://documentation.blackduck.com/bundle/detect/page/properties/deprecated-properties.html?

I am pretty sure that the answer is "yes". I was trying to confirm by running release-qa but didn't have much luck.
I looked at `GenerateDocsTask.java` and it appears to grab values from a `*-help.json` file which is generated based on this list of properties.
@sig-cpotts mentioned that he might be able to confirm this.

Reply:
@sig-cpotts confirmed this by generating the page on his end:

![image](https://github.com/user-attachments/assets/fb2d8dda-42bf-43bd-8fd9-898ed7e39e3a)